twilight-lavalink/src/node.rs

```diff
@@ -686,12 +678,12 @@ async fn backoff(
     loop {
         let request = connect_request(config)?;
 
-        match tokio_tungstenite::connect_async(request).await {
+        match request.connect().await {
             Ok((stream, response)) => return Ok((stream, response)),
             Err(source) => {
                 tracing::warn!("failed to connect to node {source}: {:?}", config.address);
 
-                if matches!(&source, TungsteniteError::Http(resp) if resp.status() == StatusCode::UNAUTHORIZED)
+                if matches!(&source, WebsocketError::Upgrade(upgrade::Error::DidNotSwitchProtocols(status)) if status == &403)
```

Comment:
Is this an intentional switch from unauthorised to a forbidden?
```suggestion
                if matches!(&source, WebsocketError::Upgrade(upgrade::Error::DidNotSwitchProtocols(status)) if status == &401)
```

Reply:
> Is this an intentional switch from unauthorised to a forbidden?

Definitely unintended, I must have overlooked that and/or confused the status codes

Reply:
Fixed